SimulationAgent/Agent.cs

```diff
@@ -240,15 +240,25 @@ private async Task SaveSimulationStatisticsAsync(IList<Simulation> simulations)
         {
             foreach (var simulation in simulations)
             {
-                DateTimeOffset now = DateTimeOffset.UtcNow;
-                TimeSpan duration = now - this.lastPolledTime;
-
-                // Save simulation statistics at specified interval
-                if (duration.Seconds >= SAVE_STATS_INTERVAL_SECS)
+                try
                 {
-                    await this.simulationManagers[simulation.Id].SaveStatisticsAsync();
+                    if (this.simulationManagers.ContainsKey(simulation.Id))
+                    {
+                        DateTimeOffset now = DateTimeOffset.UtcNow;
+                        TimeSpan duration = now - this.lastPolledTime;
+
+                        // Save simulation statistics at specified interval
+                        if (duration.Seconds >= SAVE_STATS_INTERVAL_SECS)
+                        {
+                            await this.simulationManagers[simulation.Id].SaveStatisticsAsync();
 
-                    this.lastPolledTime = now;
+                            this.lastPolledTime = now;
```

Comment:
nit: rename this var to `lastSaveStatisticsTime`

Reply:
fixed